vinyldns/resource_zone_test.go

```diff
@@ -93,19 +117,99 @@ func TestAccVinylDNSZoneWithACLAndNoRecordTypes(t *testing.T) {
 		CheckDestroy: testAccVinylDNSZoneDestroy,
 		Steps: []resource.TestStep{
 			resource.TestStep{
-				Config: testAccVinylDNSZoneConfigWithACLAndNoRecordTypes("email@foo.com"),
+				Config: testAccVinylDNSZoneConfigWithACLAndNoRecordTypes(zEmail),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVinylDNSZoneWithACLAndNoRecordTypesExists("vinyldns_zone.test_zone", zEmail),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "name", zName),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "email", zEmail),
+				),
+			},
+			resource.TestStep{
+				Config: testAccVinylDNSZoneConfigWithACLAndNoRecordTypes(zEmailUpdated),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVinylDNSZoneWithACLAndNoRecordTypesExists("vinyldns_zone.test_zone", zEmailUpdated),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "name", zName),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "email", zEmailUpdated),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "vinyldns_zone.test_zone",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateCheck:  testAccVinylDNSZoneImportStateCheck,
+			},
+		},
+	})
+}
+
+func TestAccVinylDNSZoneWithConnection(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccVinylDNSZoneDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccVinylDNSZoneConfigWithConnection(zEmail, "zone_connection"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVinylDNSZoneWithConnectionExists("vinyldns_zone.test_zone", zEmail),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "name", zName),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "email", zEmail),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "zone_connection.0.name", zConName),
+					resource.TestCheckResourceAttr("vinyldns_zone.test_zone", "zone_connection.0.key", zConKey),
```

Comment:
fun facts, when hitting a "real" environment (not docker), the key is encrypted.  I.e., it is impossible to verify the value of the key in a test against a real environment unless you know the secret and the encryption algorithm, so very likely not.

This works against docker because the default cryptography is `NoOp`